src/scheduling/dimacs_change_stats.cc

```diff
@@ -0,0 +1,42 @@
+#include "scheduling/dimacs_change_stats.h"
+
+#include "scheduling/dimacs_add_node.h"
+#include "scheduling/dimacs_remove_node.h"
+
+#include "scheduling/dimacs_new_arc.h"
+#include "scheduling/dimacs_change_arc.h"
+
+namespace firmament {
+
+template<typename Base>
+bool isInstanceOf(DIMACSChange* x) {
+	Base *base = dynamic_cast<Base *>(x);
+	return base != nullptr;
+}
+
+DIMACSChangeStats::DIMACSChangeStats(std::vector<DIMACSChange *> &changes) {
```

Comment:
Another way to do this is to have the fields (e.g. new_node, new_arc..) in the dimacs_change.cc. Then we can increment them from any subclass. Moreover, we won't have to do any instanceOf checks.